source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -91,7 +90,30 @@ Config::SharedConfig::SharedConfig(
     max_downstream_connection_duration_ = std::chrono::milliseconds(connection_duration);
   }
 
-  if (config.has_access_log_flush_interval()) {
+  if (config.has_access_log_options()) {
+    if (config.access_log_options().flush_access_log_on_connected() &&
+        config.flush_access_log_on_connected() /* deprecated */) {
+      throw EnvoyException(
+          "Only one of flush_access_log_on_connected from TcpProxy or"
+          "flush_access_log_on_connected from TcpAccessLogOptions can be specified.");
+    }
+
+    flush_access_log_on_connected_ = config.access_log_options().flush_access_log_on_connected();
+  }
+
+  flush_access_log_on_connected_ = config.flush_access_log_on_connected();
+
+  if (config.has_access_log_options() &&
+      config.access_log_options().has_access_log_flush_interval()) {
+    if (config.has_access_log_flush_interval()) {
+      throw EnvoyException("Only one of access_log_flush_interval from TcpProxy or"
+                           "access_log_flush_interval from TcpAccessLogOptions can be specified.");
+    }
+
+    const uint64_t flush_interval = DurationUtil::durationToMilliseconds(
+        config.access_log_options().access_log_flush_interval());
+    access_log_flush_interval_ = std::chrono::milliseconds(flush_interval);
+  } else if (config.has_access_log_flush_interval() /* deprecated */) {
```

Comment:
The `flush_access_log_on_connected_` will be be overwrite by the `config.flush_access_log_on_connected()`?

nit: and I think maybe we can throw exception directly when both the `access_log_options` and `access_log_flush_interval`/`flush_access_log_on_connected` are set. We needn't check the sub-field of `access_log_options`, WDYT?

Reply:
I think that's ok, but just to make sure, do we throw in case ``flush_access_log_on_connected`` (not the grouped one) is set, but 
``access_log_options().access_log_flush_interval`` is set? My point is that they are different features.

Reply:
@wbpcode I fixed the condition, but I think it would be better to keep the code layout as is because for the interval check you'd still need to check if it has a value anyways, so after trying to refactor it, the code just got more complex. Hope that's fine with you

Reply:
> I think that's ok, but just to make sure, do we throw in case `flush_access_log_on_connected` (not the grouped one) is set, but `access_log_options().access_log_flush_interval` is set? My point is that they are different features.

IMO, although they are different features, if the `access_log_options` is set, we think the new API is enabled and any deprecated API usage should be unexpected error. I am also not sure which is better. So may be defer this to the senior maintainer. cc @mattklein123

Reply:
done